### PR TITLE
[Core]Add runtime context for python worker

### PR DESCRIPTION
**Pull request by lixin-wei**
## Why are these changes needed?

Add runtime context interface for python worker. This interface already exists in java worker. This PR add a python version to keep consistency. 
## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

File: python/ray/runtime_context.py
Comment (rkooo567) [inline python/ray/runtime_context.py:1]: Unnecessary import

File: python/ray/runtime_context.py
Comment (rkooo567) [inline python/ray/runtime_context.py:26]: Let's just move this check to init.

File: python/ray/runtime_context.py
Comment (rkooo567) [inline python/ray/runtime_context.py:20]: Can you either change the method name to `driver_id` or change the docstring to use the term job_id instead of driver id?

File: python/ray/runtime_context.py
Comment (rkooo567) [inline python/ray/runtime_context.py:32]: Add TODO to make a job table?

File: python/ray/runtime_context.py
Comment (rkooo567) [inline python/ray/runtime_context.py:33]: Raise an exception instead of assert

File: python/ray/runtime_context.py
Comment (rkooo567) [inline python/ray/runtime_context.py:44]: Please write a comment for assert.

File: python/ray/runtime_context.py
Comment (rkooo567) [inline python/ray/runtime_context.py:49]: When is this useful?

File: python/ray/runtime_context.py
Comment (rkooo567) [inline python/ray/runtime_context.py:55]: Same as above?

File: python/ray/runtime_context.py
Comment (rkooo567) [inline python/ray/runtime_context.py:59]: Add TODO here instead of note.

File: python/ray/runtime_context.py
Comment (rkooo567) [inline python/ray/runtime_context.py:70]: `gcs_address`?

**Review by rkooo567 — CHANGES_REQUESTED** [review]

File: python/ray/runtime_context.py
Comment (lixin-wei) [reply python/ray/runtime_context.py:49]: currently not used, removed.

File: python/ray/runtime_context.py
Comment (lixin-wei) [reply python/ray/runtime_context.py:55]: currently not used, removed.

**Comment by rkooo567** [comment]
I think you forgot to push a new commit? I cannot see any change from the prev state.

**Comment by rkooo567** [comment]
Also, we need to add tests. Please add a simple test for each of field.

File: python/ray/runtime_context.py
Comment (rkooo567) [inline python/ray/runtime_context.py:50]: Not necessary anymore.

File: python/ray/runtime_context.py
Comment (rkooo567) [inline python/ray/runtime_context.py:52]:
```suggestion
        return actor_info and actor_info["NumRestarts"] != 0:
```

File: python/ray/runtime_context.py
Comment (rkooo567) [inline python/ray/runtime_context.py:47]: Why do we need this btw?

File: python/ray/runtime_context.py
Comment (rkooo567) [inline python/ray/runtime_context.py:31]:
```suggestion
            raise ValueError(f"Failed to get a job config from GCS for a job id, {self.current_driver_id.hex()}.")
```

File: python/ray/runtime_context.py
Comment (rkooo567) [inline python/ray/runtime_context.py:43]:
```suggestion
            assert self.worker.mode == ray.worker.WORKER_MODE, (f"This method is only available when the process is a worker. Current mode: {self.worker.mode}")
```

**Review by rkooo567 — COMMENTED** [review]
Some more comments. Since this will be user facing, we need more user-friendly error msgs.

File: python/ray/runtime_context.py
Comment (lixin-wei) [reply python/ray/runtime_context.py:47]: In streaming framework, we need this to check whether it's the first time the actor is created.

**Comment by lixin-wei** [comment]
@rkooo567 Currently only `was_current_actor_reconstructed`, `current_actor_id` and `current_job_id` will be used in streaming framework. So I removed other useless interfaces. All your suggestion is done, please review again.

**Review by rkooo567 — APPROVED** [review]

**Comment by rkooo567** [comment]
Please mark it with `test_ok` label once build passes! I will merge it after that! :)

**Pull request merged**
